src/Lykke.MatchingEngine.Connector/Services/TcpMatchingEngineClient.cs

```diff
@@ -33,16 +33,20 @@ private long GetNextRequestId()
                 return _currentNumber++;
         }
 
-        public TcpMatchingEngineClient(IPEndPoint ipEndPoint, ISocketLog socketLog = null)
+        public TcpMatchingEngineClient(IPEndPoint ipEndPoint, ISocketLog socketLog = null, bool ignoreErrors = false)
         {
             _clientTcpSocket = new ClientTcpSocket<MatchingEngineSerializer, TcpOrderSocketService>(
                 socketLog,
                 ipEndPoint,
                 3000,
                 () =>
                 {
-                    _tcpOrderSocketService = new TcpOrderSocketService(_tasksManager, _newTasksManager,
-                        _marketOrderTasksManager);
+                    _tcpOrderSocketService = new TcpOrderSocketService(
```

Comment:
Is it safe to set a field from the lambda? It can be called multiple times and every time it will update the field. Looks odd.

Reply:
I don't like this code too, but it is a constructor, and this class is used as singleton only. Anyway I would not change this code more than I need for my actual task. May be we need to create another task for refactoring.